### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by colinanderson10**
Updating requirements and adding Assignment 3 .py file

### CODE REVIEW

File: Assignment3/Assignment3.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+
+def main():
+    # Setup Spark
+    spark = SparkSession.builder.master("local[*]").getOrCreate()
+
+    # put in sql database username here
+    username = ""
+    # put in sql database password here
+    pword = ""
```
Comment (i-baumann) [inline Assignment3/Assignment3.py:18]: Not necessary, but could put an echo in the sh script letting user know to place un and pwd in the py file.

File: Assignment3/Assignment3.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+            partition = (
+                Window.partitionBy("batter")
+                .orderBy(col("gamedate").cast("timestamp").cast("long"))
+                .rangeBetween(days(-100), hours(-6))
+            )
+
+            dataset = dataset.withColumn(output_col, sum(input_cols[1]).over(partition))
+
+            return dataset
```
Comment (i-baumann) [inline Assignment3/Assignment3.py:132]: These two transformers could probably be rolled into one.

File: Assignment3/Assignment3.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+            return self._set(**kwargs)
+
+        def _transform(self, dataset):
+            output_col = self.getOutputCol()
+
+            dataset = dataset.withColumn(
+                output_col, col("rollinghits") / col("rollingatbats")
+            )
+
+            return dataset
```
Comment (i-baumann) [inline Assignment3/Assignment3.py:161]: I like how you've got this transformer separate. Like we talked about, I wish there was a way to do this without keeping the intermediate rolling hits & atbats vars (maybe there is?)

**Review by i-baumann — COMMENTED** [review]
Ran without a hitch for me, but this needs a shell script that will ensure the user's python instance has the required modules installed.

**Review by dafrenchyman — APPROVED** [review]
Please review notes in canvas

**Pull request merged**